src/ts/components/extensions/richtexteditor/RichTextEditor.tsx

```diff
@@ -0,0 +1,96 @@
+import { RichTextEditorLabels } from "@mantine/tiptap";
+import { BoxProps } from "props/box";
+import { DashBaseProps } from "props/dash";
+import { StylesApiProps } from "props/styles";
+import React, { Suspense } from "react";
+
+// eslint-disable-next-line no-inline-comments
+const LazyRichTextEditor = React.lazy(() => import(/* webpackChunkName: "RichTextEditor" */ './fragments/RichTextEditor'));
+
+/** Supported extensions (add more as needed). */
+type ExtensionName =
+  | "StarterKit"
+  | "Underline"
+  | "Link"
+  | "Superscript"
+  | "Subscript"
+  | "Highlight"
+  | "TextAlign"
+  | "TextStyle"
+  | "Table"
+  | "TableCell"
+  | "TableRow"
+  | "TableHeader"
+  | "Placeholder"
+  | "Image"
+  | "Color";
+
+// TODO: Maybe add types for the extensions options explicitly?
+type Extension =
+  | ExtensionName
+  | { [key in ExtensionName]?: Record<string, unknown> }; // Allow any options
+
+/** Supported controls (add more as needed). */
+type ControlName = "Bold" | "Italic" | "Underline" | "Strikethrough" | "ClearFormatting" | "Highlight" | "Code" | "H1" | "H2" | "H3" | "H4" | "Blockquote" | "Hr" | "BulletList" | "OrderedList" | "Subscript" | "Superscript" | "Link" | "Unlink" | "AlignLeft" | "AlignCenter" | "AlignJustify" | "AlignRight" | "Undo" | "Redo" | "Color" | "ColorPicker" | "UnsetColor";
```

Comment:
missing H5, H6, CodeBlock

Reply:
Nicely spotted! Thanks! They should be there now